vlm/src/main/scala/geotrellis/contrib/vlm/geotiff/GeoTiffResampleRasterSource.scala

```diff
@@ -30,12 +30,13 @@ case class GeoTiffResampleRasterSource(
   resampleGrid: ResampleGrid[Long],
   method: ResampleMethod = NearestNeighbor,
   strategy: OverviewStrategy = AutoHigherResolution,
-  private[vlm] val targetCellType: Option[TargetCellType] = None
+  private[vlm] val targetCellType: Option[TargetCellType] = None,
+  private val baseTiff: Option[MultibandGeoTiff] = None
```

Comment:
I wonder if this is actually something we need for all `GeoTiff` constructors - it seems to me like only the base `RasterSource` should have/needs this cached tiff

Reply:
That wouldn't work as the base `RasterSource` doesn't contain any information regarding the format of the data (ie GeoTiffs, GDALDatasets, etc), but rather, just describes it; so we wouldn't be able to put caching logic there.

Reply:
I'm not sure I follow. It appears to me that no extra logic is being done to modify `baseTiff`. My suggestion would be to simply remove `baseTiff` from all constructors other than the base (the `GeoTiffRasterSource` in this case)

Reply:
Oh, I misunderstood. I thought you were referring to the `RasterSource` trait.

The problem with that is neither `GeoTiffReprojectRasterSoure` nor `GeoTiffResampleRasterSource` extends `GeoTiffRasterSource`. They're each a separate from one another. Also, there can be instances where the user creates either a `GeoTiffReprojectRasterSoure` or a `GeoTiffResampleRasterSource` directly, instead of first making a `GeoTiffRasterSource` and then doing some kind of transformation.

Reply:
Misunderstandings all around... I incorrectly thought (this may have changed recently) that these `RasterSource` operations still involved encoding transformations via case class nesting.

LGTM in that case